apps/emqx_bridge_gcp_pubsub/src/emqx_bridge_gcp_pubsub_consumer_worker.erl

```diff
@@ -0,0 +1,552 @@
+    handle_continue/2,
+    terminate/2
+]).
+
+-export([get_subscription/1]).
+-export([reply_delegator/3, pull_async/1, process_pull_response/2, ensure_subscription/1]).
+
+-type subscription_id() :: binary().
+-type bridge_name() :: atom() | binary().
+-type ack_id() :: binary().
+-type config() :: #{
+    ack_retry_interval := emqx_schema:timeout_duration_ms(),
+    connector_state := emqx_bridge_gcp_pubsub_connector:state(),
+    ecpool_worker_id => non_neg_integer(),
+    hookpoint := binary(),
+    instance_id := binary(),
+    mqtt_config => emqx_bridge_gcp_pubsub_impl_consumer:mqtt_config(),
+    pull_max_messages := non_neg_integer(),
+    subscription_id => subscription_id(),
+    topic => emqx_bridge_gcp_pubsub_connector:topic()
+}.
+-type state() :: #{
+    ack_retry_interval := emqx_schema:timeout_duration_ms(),
+    ack_timer := undefined | reference(),
+    async_workers := #{pid() => reference()},
+    connector_state := emqx_bridge_gcp_pubsub_connector:state(),
+    ecpool_worker_id := non_neg_integer(),
+    hookpoint := binary(),
+    instance_id := binary(),
+    mqtt_config => emqx_bridge_gcp_pubsub_impl_consumer:mqtt_config(),
+    pending_acks => [ack_id()],
+    pull_max_messages := non_neg_integer(),
+    subscription_id => subscription_id(),
+    topic => emqx_bridge_gcp_pubsub_connector:topic()
+}.
+-type decoded_message() :: map().
+
+-define(HEALTH_CHECK_TIMEOUT, 10_000).
+-define(OPTVAR_SUB_OK(PID), {?MODULE, PID}).
+
+%%-------------------------------------------------------------------------------------------------
+%% API used by `reply_delegator'
+%%-------------------------------------------------------------------------------------------------
+
+-spec pull_async(pid()) -> ok.
+pull_async(WorkerPid) ->
+    gen_server:cast(WorkerPid, pull_async).
+
+-spec process_pull_response(pid(), binary()) -> ok.
+process_pull_response(WorkerPid, RespBody) ->
+    gen_server:cast(WorkerPid, {process_pull_response, RespBody}).
+
+-spec ensure_subscription(pid()) -> ok.
+ensure_subscription(WorkerPid) ->
+    gen_server:cast(WorkerPid, ensure_subscription).
+
+-spec reply_delegator(pid(), binary(), {ok, map()} | {error, timeout | term()}) -> ok.
+reply_delegator(WorkerPid, InstanceId, Result) ->
+    case Result of
+        {error, timeout} ->
+            ?MODULE:pull_async(WorkerPid);
+        {error, Reason} ->
+            ?SLOG(warning, #{
+                msg => "gcp_pubsub_consumer_worker_pull_error",
+                instance_id => InstanceId,
+                reason => Reason
+            }),
+            case Reason of
+                #{status_code := 409} ->
+                    %% the subscription was not found; deleted?!
+                    ?MODULE:ensure_subscription(WorkerPid);
+                _ ->
+                    ?MODULE:pull_async(WorkerPid)
+            end;
+        {ok, #{status_code := 200, body := RespBody}} ->
+            ?MODULE:process_pull_response(WorkerPid, RespBody)
+    end.
```

Comment:
It's a bit difficult to trace that all paths finally renew `pull_async`. Also, we must ensure that a call to `reply_delegator` should never be lost.

Maybe we also add some timeout-based `pull_async`?

Reply:
:heavy_check_mark: